admin/js/wizard.js

```diff
@@ -119,7 +121,54 @@ function setOption(option) {
       showMessage(`Error setting option: ${option.key}`, 'error');
     });
 }
-
+function onClickOptionButton(event) {
+  if (event){
+    event.preventDefault();
+  }
+  var formData = new FormData(event.target, event.submitter);
+  const button = formData.get('button');
+  if (button !== 'all') {
+  	try {
+      setOption(
+        {
+          "key":button,
+          "value":formData.get(button)
+        }
+      );
+    } catch (error) {
+      console.error(error);
+      showMessage('Error setting option', 'error');
+    }
+  } else{
+    for (entry of formData.entries()) {
+      try {
+        if (entry.value !== 'all') {
+          setOption(
+            {
+              "key":entry[0],
+              "value":entry[1]
+            }
+          );
+        }
+      } catch (error) {
+        console.error(error);
+        showMessage('Error setting option', 'error');
+      }
+    }
```

Comment:
Javascript has a set of array functions that could make this a little easier: map, filter, and reduce. Here's some general info about them: https://www.freecodecamp.org/news/javascript-map-reduce-and-filter-explained-with-examples/

We can use map here to change the structure of the entries array into the object structure we need. We can also first use filter to exclude "all" from the result:

```
const options = formData.entries()
  .filter((entry) => entry[0] !== 'button')
  .map((entry) => ({
    key: entry[0],
    value: entry[1],
  });
```

You can see that these can be chained together. Each of them returns the modified array from its operation. We could then add on the `forEach()` method to set the option:

```
formData.entries()
  .filter((entry) => entry[0] !== 'button')
  .map((entry) => ({
    key: entry[0],
    value: entry[1],
  })
  .forEach((option) => setOption(option);
```

This does the same thing as what you've written, but you'll notice that there is less nesting which makes it a bit easier to read and follow what is happening.